File: reverse-integer/reverse-integer.py
class Solution:
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        mark = 1
        if x < 0:
            mark = -1
            x *= -1
        res = mark * int(str(x)[::-1])
        if x < -2**31 and x > 2**31 - 1:
            return 0
        return res

    def reverse2(self, x):
        # 记录正负
        mark = 1
        if x < 0:
            mark = -1
        # 利用绝对值函数、整型转字符串函数，然后进行反转，添加记录的正负符号
        res = mark * int(str(abs(x))[::-1])
        # 返回反转值，超出给定范围为 0
        return res if -(2**31) - 1 < res < 2**31 else 0


if __name__ == "__main__":
    print(Solution().reverse2(123))  # 321
    print(Solution().reverse2(-123))  # -321
    print(Solution().reverse2(120))  # 21

File: reverse-integer/reverse-integer.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function (x) {
    const minn = -(1 << 30) * 2,
        maxn = (1 << 30) * 2 - 1,
        arr = x.toString().split('')
    let res = ""
    if (x < 0) {
        arr.shift()
    }
    res = Number(arr.reverse().join(''))
    if (x < 0) {
        res *= -1
    }
    if (res < minn || res > maxn) {
        return 0
    } else {
        return res
    }
};

var reverse2 = function (x) {
    const minn = Math.pow(-2, 31),
        maxn = Math.pow(2, 31) - 1,
        arr = Math.abs(x).toString().split("")
    let res = ""
    console.log(minn, maxn);
    for (let i = arr.length - 1; i >= 0; i--) {
        res += arr[i]
    }
    if (x > 0 && +res <= maxn) {
        return +res
    }
    if (x < 0 && -res >= minn) {
        return -res
    }
    return 0
};

// 测试
console.log(reverse2(-123)) // -321
console.log(reverse2(120)) // 21